userspace/libsinsp/threadinfo.cpp

```diff
@@ -876,7 +876,7 @@ double sinsp_threadinfo::get_fd_usage_pct_d()
 
 uint64_t sinsp_threadinfo::get_fd_opencount() const
 {
-	return get_main_thread()->m_fdtable.size();
+	return ((sinsp_threadinfo*)this)->get_main_thread()->m_fdtable.size();
```

Comment:
casting "this" to the type that it should already be? this smells fishy. what is the error you get when you don't have this?

Reply:
yes, I think in the case of windows, you declared it as a non-const function, presumably because of the "return this;" on line 998, which happens if we ARE the main thred, in which case, stripping the const decorator is somewhat reasonable.

the linux version of this solves it by having the const_cast exactly where it's needed, on threadinfo.h:137.

So the issue here is that get_main_thread is const on linux and not on windows. Given that the fucntions are otherwise identical, why can't we use the same version of the function? or that failing, why should we have a const implementation for linux and a non-const one for windows? Though I THINK the const one should satisfy both use cases (I don't have a windows setup to try), at worst case, as the "const" decorator changes the function signature, it would be not incorrect to have both implementations available on both platforms.